### PR TITLE
Prevent same email registration.

### PR DESCRIPTION
**Pull request by peregrineshahin**

### CODE REVIEW

**Comment by vdbergh** [comment]
```
    result = request.userdb.create_user(
        username=signup_username, password=signup_password, email=validated_email
    )
    if not result:
        request.session.flash("Error! Invalid username or password", "error")
```
This error message in `signup()` should be adapted.

**Comment by peregrineshahin** [comment]
Done
> ```
>     result = request.userdb.create_user(
>         username=signup_username, password=signup_password, email=validated_email
>     )
>     if not result:
>         request.session.flash("Error! Invalid username or password", "error")
> ```
> 
> This error message in `signup()` should be adapted.

**Comment by vdbergh** [comment]
If the Fishtest server could send email (i.e. if the DNS issues are sorted out) then it would be possible to send the user a confirmation link.

Then the life cycle of the user state would be
```
pending-->confirmed-->accepted-->blocked-->unblocked-->blocked-->...
```

**Comment by peregrineshahin** [comment]
indeed, a confirmation email is neccessary IMO. We use the email for contacting concerning worker issues and we should know that we are not sending someone else.

File: server/fishtest/userdb.py
Comment (vdbergh) [inline server/fishtest/userdb.py:123]: This will not work no? `create_user()` does not know about `request`. Besides `userdb` is for handling the db, not for user interaction.

Comment (peregrineshahin) [reply server/fishtest/userdb.py]: > This will not work no? `create_user()` does not know about `request`. Besides `userdb` is for handling the db, not for user interaction.

Please check now

**Comment by peregrineshahin** [comment]
Does that replace manual accepting users?

> If the Fishtest server could send email (i.e. if the DNS issues are sorted out) then it would be possible to send the user a confirmation link.
> 
> Then the life cycle of the user state would be
> 
> ```
> pending-->blocked-->unblocked-->blocked-->unblocked-->blocked-->...
> ```
> 
> EDIT: I shortened the life cycle even more. The transition pending-->blocked should happen after the confirmation link has been clicked.

File: server/fishtest/userdb.py
Comment (vdbergh) [reply server/fishtest/userdb.py]: This seems ok!

**Comment by vdbergh** [comment]
> Does that replace manual accepting users?
> 
That's a policy choice. One could decide that after receiving the confirmation the user is automatically unblocked. Or one could require manual intervention by an approver to unblock the user.

**Review by ppigazzini — APPROVED** [review]
Looks good on DEV

**Comment by ppigazzini** [comment]
PROD updated, thank you @peregrineshahin :)

**Pull request merged**
